services/researchAgentService.ts

```
@@ -88,49 +106,98 @@ export class ResearchAgentService {
-        } catch (error) {
-          if (error instanceof APIError && error.code === 'AUTH_ERROR') {
-            // Fallback to Gemini
-            console.warn('Grok API key not available, falling back to Gemini');
-            return this.generateTextWithGemini(prompt, GENAI_MODEL_FLASH, effort, useSearch);
+    try {
+      switch (model) {
+        case GENAI_MODEL_FLASH:
+          const geminiResult = await geminiService.generateContent(prompt, {
+            model: GENAI_MODEL_FLASH,
+            useSearch: true
+          });
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)